src/game_config_manager.cpp

```diff
@@ -471,6 +471,30 @@ void game_config_manager::load_addons_cfg()
 					cfg["addon_version"] = addon_version.str();
 				}
 			}
+			config advancefroms;
+			for(auto& units : umc_cfg.child_range("units")) {
+				for(auto& unit_type : units.child_range("unit_type")) {
+					for(const auto& advancefrom : units.child_range("advancefrom")) {
+
+						deprecated_message(
+							_("[advancefrom]"),
+							DEP_LEVEL::FOR_REMOVAL,
+							{1, 17, 0},
+							_("Use [modify_unit_type] instead")
```

Comment:
To make this really easy for someone to update their WML it’d be nice if this outputs the exact tag to use. I assume it’s difficult for wmllint to rewrite the WML since it‘d need to figure out where the campaign tag is but it could output the exact replacement as well, of course.